docs/FAQ.md

```diff
@@ -0,0 +1,121 @@
+
+TODO: list equivalent of some common macros
+
+### Is the nightly compiler needed for some features?
+
+Currently, the `interface` and `implementation` proc macros (used to define and implement COM interfaces) require some nightly features. However, those features are in the process of being stablized and so the `interface` and `implementation` macros should soon be available on stable Rust. See [this issue](https://github.com/microsoft/windows-rs/issues/1523) for more information.
+
+## COM
+
+### How to query for a specific COM interface?
+
+While raw COM uses the `QueryInterface` method for querying for an interface, the `windows` crate provides the safe `cast` method as an alternative.
+
+```rust
+ // Assume `a: IStringable`
+ let b: IClosable = a.cast()?;
+```
+
+The cast method returns a Rust `Result` type allowing you to handle calls to `cast` just like you would any other fallible Rust function.
+
+### How do I declare a new COM interface?
+
+The `windows` crate comes with definitions for all the interfaces you're likely to encounter when using the Windows SDK, but you might still have the need to define an interface yourself.
+
+To do so, you'll need to use the `interface` feature which (like any Cargo feature) can be enabled in your project's Cargo.toml file.
+
+```toml
+windows = { version = "..", features = ["interface"] }
+```
+
+Then inside your project you define the interface in much the same way you define a Rust crate just with the `#[windows::interface]` annotation above the trait definition.
+
+```rust
+/// My interface
+#[windows::core::interface("BD1AE5E0-A6AE-11CE-BD37-504200C10000")]
+unsafe trait IMyInterface: IUnknown {
+    unsafe fn MyFunction(&self) -> windows::core::HRESULT;
+}
+```
+
+### How do I implement an existing COM interface?
+
+If you need to implement a COM interface for a type, you'll need to add the `implement` feature which (like any Cargo feature) can be enabled in your project's Cargo.toml file.
+
+### How are optional COM interfaces handled?
+
+```toml
+windows = { version = "..", features = ["implement"] }
+```
+
+Then you'll need to declare that your type implements a particular interface by adding the `#[implement]` proc macro to your type and then writing an `impl` block for the interface. For a trait called `IMyTrait` you will need to implement the `IMyTrait_Impl` trait (note the trailing `_Impl` in the name).
+
+```rust
+#[windows::core::implement(IMyInterface)]
+struct MyStruct;
+
+impl IMyInterface_Impl for MyStruct {
+    fn MyMethod(&self) -> windows::core::HRESULT {
+        todo!("Your implementation goes here");
+    }
+}
+
+```
+
+## Translating the Windows API into Rust
```

Comment:
Maybe this? The existing title sounds like it'll be a section on building crates from scratch (which is documentation I'm working on).

```suggestion
## Calling Windows APIs
```